cli/cmd/generate_oci.go

```diff
@@ -0,0 +1,446 @@
+		if iacParamsExpired {
+			cli.Log.Debug("loaded previously set values for OCI iac generation")
+		}
+
+		extraState := &OciGenerateCommandExtraState{}
+		extraStateParamsExpired := cli.ReadCachedAsset(CachedAssetOciExtraState, &extraState)
+		if extraStateParamsExpired {
+			cli.Log.Debug("loaded previously set values for OCI iac generation (extra state)")
+		}
+
+		// Determine if previously cached options exists; prompt user if they'd like to continue
+		answer := false
+		if !iacParamsExpired || !extraStateParamsExpired {
+			if err := SurveyQuestionInteractiveOnly(SurveyQuestionWithValidationArgs{
+				Prompt:   &survey.Confirm{Message: QuestionUsePreviousCache, Default: false},
+				Response: &answer,
+			}); err != nil {
+				return errors.Wrap(err, "failed to load saved options")
+			}
+		}
+
+		// If the user decides NOT to use the previous values; we won't load them.  However, every time the command runs
+		// we are going to write out new cached values, so if they run it - bail out - and run it again they'll get
+		// re-prompted.
+		if answer {
+			// Merge cached inputs to current options (current options win)
+			if err := mergo.Merge(GenerateOciCommandState, cachedOptions); err != nil {
+				return errors.Wrap(err, "failed to load saved options")
+			}
+			if err := mergo.Merge(GenerateOciCommandExtraState, extraState); err != nil {
+				return errors.Wrap(err, "failed to load saved options")
+			}
+		}
+	}
+
+	// Collect and/or confirm parameters
+	err = promptOciGenerate(GenerateOciCommandState, GenerateOciCommandExtraState)
+	if err != nil {
+		return errors.Wrap(err, "collecting/confirming parameters")
+	}
+
+	return nil
+}
+
+func initGenerateOciTfCommandFlags() {
+	// add flags to sub commands
+	generateOciTfCommand.PersistentFlags().BoolVar(
+		&GenerateOciCommandState.Config,
+		"config",
+		false,
+		"enable config integration")
+	generateOciTfCommand.PersistentFlags().StringVar(
+		&GenerateOciCommandState.ConfigName,
+		"config_name",
+		"",
+		"specify name of config integration")
+	generateOciTfCommand.PersistentFlags().StringVar(
+		&GenerateOciCommandState.TenantOcid,
+		"tenant_ocid",
+		"",
+		"specify the OCID of the tenant to integrate")
+	generateOciTfCommand.PersistentFlags().StringVar(
+		&GenerateOciCommandState.OciUserEmail,
+		"oci_user_email",
+		"",
+		"specify the email of the OCI user created for integration")
+	generateOciTfCommand.PersistentFlags().BoolVar(
+		&GenerateOciCommandExtraState.TerraformApply,
+		"apply",
+		false,
+		"run terraform apply without executing plan or prompting",
+	)
+	generateOciTfCommand.PersistentFlags().StringVar(
+		&GenerateOciCommandExtraState.Output,
+		"output",
+		"",
+		"location to write generated content (default is ~/lacework/oci)",
+	)
+}
+
+// basic validation of Tenant OCID format
+func validateOciTenantOcid(val interface{}) error {
+	return validateStringWithRegex(
+		val,
+		`ocid1\.tenancy\.[^\.\s]*\.[^\.\s]*(\.[^\.\s]+)?\.[^\.\s]+`,
```

Comment:
Hazarding a guess this may have been pulled from some OCI doc somewhere. If so, can we add a comment linking to the doc?

Reply:
done!

Reply:
Not seeing this change reflected at the moment. Perhaps it hasn't been pushed yet?

Reply:
How about now?